python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1209,3 +1209,113 @@ def empty_subset(
         cls, partitions_def: Optional[PartitionsDefinition] = None
     ) -> "DefaultPartitionsSubset":
         return cls()
+
+
+class AllPartitionsSubset(
+    NamedTuple(
+        "_AllPartitionsSubset",
+        [
+            ("partitions_def", PartitionsDefinition),
+            ("dynamic_partitions_store", Optional[DynamicPartitionsStore]),
+        ],
+    ),
+    PartitionsSubset,
+):
+    """This is an in-memory (i.e. not serializable) convenience class that represents all partitions
+    of a given PartitionsDefinition, allowing set operations to be taken without having to load
+    all partition keys immediately.
+    """
+
+    def __new__(
+        cls,
+        partitions_def: PartitionsDefinition,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore],
+    ):
+        return super().__new__(
+            cls, partitions_def=partitions_def, dynamic_partitions_store=dynamic_partitions_store
+        )
+
+    def get_partition_keys(self, current_time: Optional[datetime] = None) -> Iterable[str]:
+        return self.partitions_def.get_partition_keys(current_time, self.dynamic_partitions_store)
+
+    def get_partition_keys_not_in_subset(
+        self,
+        partitions_def: PartitionsDefinition,
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> Iterable[str]:
+        return set()
+
+    def get_partition_key_ranges(
+        self,
+        partitions_def: PartitionsDefinition,
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> Sequence[PartitionKeyRange]:
+        first_key = partitions_def.get_first_partition_key(current_time, dynamic_partitions_store)
+        last_key = partitions_def.get_last_partition_key(current_time, dynamic_partitions_store)
+        if first_key and last_key:
+            return [PartitionKeyRange(first_key, last_key)]
+        return []
+
+    def with_partition_keys(self, partition_keys: Iterable[str]) -> "AllPartitionsSubset":
+        return self
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, AllPartitionsSubset) and other.partitions_def == self.partitions_def
+        )
+
+    def __and__(self, other: "PartitionsSubset") -> "PartitionsSubset":
+        return other
+
+    def __sub__(self, other: "PartitionsSubset") -> "PartitionsSubset":
+        if self == other:
+            return self.partitions_def.empty_subset()
+        return self.partitions_def.empty_subset().with_partition_keys(
+            set(self.get_partition_keys()).difference(set(other.get_partition_keys()))
+        )
+
+    def __or__(self, other: "PartitionsSubset") -> "PartitionsSubset":
+        return self
+
+    def __len__(self) -> int:
+        return len(
+            self.partitions_def.subset_with_all_partitions(
```

Comment:
nit: can we just return `len(self.partitions_def.get_partition_keys())` instead?

Reply:
yep good point